### PR TITLE
Change sec2tick macros to round up

### PR DESCRIPTION
**Pull request by jlaitine**
## Summary

This change is discussed in https://github.com/apache/nuttx/issues/8346 ; in short, this proposes changing USEC2TICK, MSEC2TICK and SEC2TICK to round up instead of round to nearest. This is the same practice that is used in many other OSs (like linux msecs_to_jiffies ; when converting a value to ticks the integer result is always greater or equivalent of the exact result. 

The rationale behind this change is that typical use is a timeout or schedule wq event. That is, to go to sleep for *at least* N msec.

## Impact

This fixes regression in i2c drivers, networking and iob caused by https://github.com/apache/nuttx/commit/1fb8c13e5e80ee4cd5758e76821378162451c2fa

## Testing

It has been tested that the change fixes the found problems on stm32f7 and mpfs targets with 10ms systick. However, this affects many drivers in many platforms, I am not able to test them all.

### CODE REVIEW

**Comment by btashton** [comment]
This looks reasonable to me, but it would be really good to validate that we have not broken anything on a few different core platforms. Maybe not prior to merging but at least prior to the next release.

File: include/nuttx/clock.h [hunk 1]
```diff
@@ -145,41 +145,49 @@
...
-#define TICK_PER_MIN          (USEC_PER_MIN  / USEC_PER_TICK)            /* Truncates! */
-#define TICK_PER_SEC          (USEC_PER_SEC  / USEC_PER_TICK)            /* Truncates! */
-#define TICK_PER_MSEC         (USEC_PER_MSEC / USEC_PER_TICK)            /* Truncates! */
-#define TICK_PER_DSEC         (USEC_PER_DSEC / USEC_PER_TICK)            /* Truncates! */
-#define TICK_PER_HSEC         (USEC_PER_HSEC / USEC_PER_TICK)            /* Truncates! */
+/* TICK_PER_* truncates! */
 
-#define MSEC_PER_TICK         (USEC_PER_TICK / USEC_PER_MSEC)            /* Truncates! */
-#define NSEC_PER_TICK         (USEC_PER_TICK * NSEC_PER_USEC)            /* Exact */
+#define TICK_PER_HOUR         (USEC_PER_HOUR / USEC_PER_TICK)
```
Comment (xiaoxiang781216) [inline include/nuttx/clock.h:150]: should we use the roundup for these constant too?

Comment (jlaitine) [reply include/nuttx/clock.h:150]: Intuitively I'd say no:

- The intended use for those macros is different than of those that I was now changing
- There is a comment warning about these macros a bit above, describing the truncation/inaccuracy
- They haven't been used in many places (basically only TICK_PER_SEC is used) and in most (normal) cases it wouldn't make any difference either
- The TICK_PER_SEC is used in clock_ticks2time.c in a similar way as some other OSs do tick->timespec conversion, so it is currenly expecting it to truncate.
- They are also not directly related to the SEC2TICK macros, as due to rounding it is not anyway possible to do 1<->1 conversion between those, regardless of what rounding is used

So I wouldn't touch them now unless there is a good reason to do so (?)

Comment (xiaoxiang781216) [reply include/nuttx/clock.h:150]: Ok.

**Review by xiaoxiang781216 — APPROVED** [review]

**Review by pkarashchenko — APPROVED** [review]
Looks good, but need to check all places that use sleep since `usleep(X);` will sleep at least X + 1 ticks so extra tick will be added if number of ticks are calculated with this macro

**Pull request merged**